src/Core/CrestApps.OrchardCore.AI.Core/Services/DefaultAIChatSessionManager.cs

```diff
@@ -134,4 +134,49 @@ public Task SaveAsync(AIChatSession chatSession)
 
         return _session.SaveAsync(chatSession, collection: AIConstants.CollectionName);
     }
+
+    public async Task<bool> DeleteAsync(string sessionId)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(sessionId);
+
+        var chatSession = await FindAsync(sessionId);
```

Comment:
Don't use `FindAsync`. Instead do something like this

```
var user = _httpContextAccessor.HttpContext?.User;

if (user.Identity?.IsAuthenticated == true)
{
    var userId = user.FindFirstValue(ClaimTypes.NameIdentifier);

    var session = await _session.Query<AIChatSession, AIChatSessionIndex>(i => i.SessionId == sessionId && i.UserId == userId, collection: AIConstants.CollectionName)
        .FirstOrDefaultAsync();
}
```

This way we only delete a session that is tied to the userId only.